cpp/tests/utilities/column_utilities.cu

```diff
@@ -46,78 +47,310 @@
 #include "rmm/cuda_stream_view.hpp"
 
 namespace cudf {
+
 namespace test {
 
 namespace {
 
+// expand all non-null rows in a list column into a column of child row indices.
+std::unique_ptr<column> generate_child_row_indices(lists_column_view const& c,
+                                                   column_view const& row_indices)
+{
+  // Example input
+  // List<int32_t>:
+  // Length : 7
+  // Offsets : 0, 3, 6, 8, 11, 14, 16, 19
+  //                 |     |                        <-- non-null input rows
+  // Null count: 7
+  // 0010100
```

Comment:
I might be misunderstanding the example here. Should this not be `5`?
```suggestion
  // Null count: 5
  // 0010100
```

Reply:
The example Dave talked in Slaced has all nulls :smile: